cards.py

```diff
@@ -45,33 +45,35 @@ def get(self, id):
 			self.error(404)
 			return
 		
-		# TODO: Replace with fetching the current semester once semester formatting functions are merged in.
-		semester = 'Fall 2017'
-		# Replace 0 with O because it looks better in the font.
-		semester_mod = semester.replace('0', 'O')
-		
+		# Pick current semester by default, otherwise if there is a parameter in the request use that as a reference
+		semester = validate_semester(self.request.get('semester'))
+		if not semester:
+			semester = get_current_semester()
+
 		template_vals = {
 			'title': 'ID card for ' + member.name,
 			'members': [member],
-			'semester': semester_mod
+			# Replace 0 with O because it looks better in the font
+			'semester': semester_pretty(semester).replace('0', 'O')
```

Comment:
While this is less lines, the `semester_mod` (you are welcome to rename the variable if you so desire) line was intentionally separated out to be less confusing to future maintainers of the code (and it avoids having a comment in the middle of a dictionary literal).

Reply:
After IRL discussion at today's meeting, disregard.